Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -466,6 +466,7 @@ - (void)flushWithMaxSize:(NSUInteger)maxBatchSize
     [self dispatchBackground:^{
         if ([self.queue count] == 0) {
             SEGLog(@"%@ No queued API calls to flush.", self);
+            [self endBackgroundTask];
```

Comment:
The `return;` on the following line prevents the batch request from ever starting. The other two `endBackgroundTask` calls are in the completion handler of the batch request.